control-plane/agents/src/bin/core/controller/registry.rs

```diff
@@ -501,4 +501,26 @@ impl Registry {
             false => vec![],
         }
     }
+
+    /// Register a frontend node(ex:- csi node) to control plane.
+    pub(crate) async fn register_frontend_node_spec(
+        &self,
+        frontend_node_spec: &RegisterFrontendNode,
+    ) -> Result<(), SvcError> {
+        self.specs()
+            .register_frontend_node_spec(self, frontend_node_spec)
+            .await?;
+        Ok(())
+    }
+
+    /// Deregister Register a frontend node(ex:- csi node) from control plane.
+    pub(crate) async fn deregister_frontend_node_spec(
```

Comment:
Name is quite long, probably can just say de/register frontend node without spec.

```suggestion
    /// Deregister a frontend node (ex: a csi node) from the control-plane.
    pub(crate) async fn deregister_frontend_node(
```

Reply:
Same reason for putting the word spec.